ESSIFI-170 Add class-validator to pex-ms

zoemaas opened the pull request:
- Added class-validator to pex-ms;
   - Fixed bugs related to the entity id;
   - Using POST instead of PUT to update the status of the PE;

Commit:
ESSIFI-170 Add class-validator to pex-ms

   - Added class-validator to pex-ms;
   - Fixed bugs related to the entity id;
   - Using POST instead of PUT to update the status of the PE;

Commit:
ESSIFI-170 Add class-validator to pex-ms

   - Updated ormconfig.json;

Commit:
Merge branch 'develop' of github.com:Sphereon-Opensource/pex-ms into feature/ESSIFI-170

 Conflicts:
	src/entity/presentationDefinition/presentationDefinitionWrapperEntity.ts

Commit:
ESSIFI-170 Add class-validator to pex-ms

   - Fixed merging issues;
   - Removed commented code;

Commit:
Merge branch 'develop' of github.com:Sphereon-Opensource/pex-ms into feature/ESSIFI-170

 Conflicts:
	src/controllers/presentationController.ts

Commit:
ESSIFI-60 Add class-validator to pex-ms

   - Added class-validator;
   - Ran eslint fix;
   - Improved error handling;
   - Status created and submitted automatically handled;

Commit:
ESSIFI-170 Add class-validator to pex-ms

   - Moved the status to its own class;
   - Added challenge token and thread id validation in POST endpoints;
   - Refactoring of helper methods;

nklomp commented on package.json:
```diff
@@ -16,6 +16,9 @@
   "dependencies": {
     "@sphereon/pex": "^0.6.3-unstable.1",
```
0.7.0 is out, which should not really impact the integration, but contains several fixes

zoemaas replied:
I will update it.

nklomp commented on src/entity/presentation/verifiableCredentialEntity.ts:
```diff
@@ -1,36 +1,60 @@
+  @Column()
+  @IsNotEmpty({ message: 'VerifiableCredential.id is invalid' })
+  //@ts-ignore
+  id: string;
```
PK

nklomp commented on src/entity/presentation/presentationWrapperEntity.ts:
```diff
@@ -1,20 +1,54 @@
+  @Column()
+  @IsNotEmpty({ message: 'PresentationDefinitionWrapper.id is invalid' })
+  // @ts-ignore
+  id: string;
```
PK

zoemaas replied:
That one needs to match the presentation id afaik, so I will leave it as it is for the same reasons as above.

nklomp commented on src/entity/presentation/presentationSubmissionEntity.ts:
```diff
@@ -1,14 +1,21 @@
+
   @Column()
+  @IsNotEmpty({ message: 'PresentationSubmission.id is invalid' })
   id: string;
```
PK

zoemaas replied:
Embedded document

nklomp commented on src/entity/presentation/presentationEntity.ts:
```diff
@@ -1,22 +1,34 @@
-    proof?: IProof | IProof[];
-}
\ No newline at end of file
+  @Column()
```
Missing an autogenerated id that is PK

zoemaas replied:
Embedded document

nklomp commented on src/entity/presentationDefinition/presentationDefinitionWrapperEntity.ts:
```diff
@@ -14,14 +16,17 @@ export class PresentationDefinitionWrapperEntity {
   @Column()
+  @IsNotEmpty({ message: 'PresentationDefinitionWrapper.id is invalid' })
   // @ts-ignore
   id: string;
```
PK

zoemaas replied:
That one needs to match the presentation id afaik, so I will leave it as it is

nklomp replied:
Primary keys do not have to be autogenerated, they need to be unique (and in most cases are autogenerated).

nklomp reviewed (changes requested):
I probably should have more remarks, but given the deadline ok for the rest.
Missing some PKs and would update to pex 0.7.0

Commit:
ESSIFI-170 Add class-validator to pex-ms

   - Created pd status and presentation status classes;
   - Created presentationDefinitionStatus.ts;
   - Improved error handling;

Commit:
Merge remote-tracking branch 'origin/feature/ESSIFI-170' into feature/ESSIFI-170

# Conflicts:
#	src/controllers/presentationController.ts

Commit:
ESSIFI-170 Add class-validator to pex-ms

   - Removed status controller;

nklomp reviewed (approved):
According to offline discussion, lets continue
